pandas/core/arrays/arrow/array.py

```diff
@@ -1209,6 +1209,16 @@ def _mode(self: ArrowExtensionArrayT, dropna: bool = True) -> ArrowExtensionArra
         """
         if pa_version_under6p0:
             raise NotImplementedError("mode only supported for pyarrow version >= 6.0")
+
+        pa_type = self._data.type
+        if pa.types.is_temporal(pa_type):
+            nbits = pa_type.bit_width
+            dtype = f"int{nbits}[pyarrow]"
+            obj = cast(ArrowExtensionArrayT, self.astype(dtype, copy=False))
```

Comment:
Can we just `self._data.cast` to int type, continue the logic below, and then `most_common.cast` back?

Reply:
we could.  i preferred this way since it was self-contained

Reply:
I suspect going through `pa.array.cast` may be more performant than `astype`. Mind checking?

Reply:
updated + green

Reply:
yes, in some cases this needs an int32 and in some cases an int64, which makes the .cast version extra-complicated